centreon/tests/e2e/features/Cloud-notifications/04-notification-edit/Index.ts

```diff
@@ -0,0 +1,266 @@
+    template: 'generic-host'
+  })
+    .addService({
+      activeCheckEnabled: false,
+      host: data.hosts.host1.name,
+      maxCheckAttempts: 1,
+      name: data.services.service1.name,
+      template: 'Ping-LAN'
+    })
+    .applyPollerConfiguration();
+
+  checkHostsAreMonitored([
+    {
+      name: data.hosts.host1.name
+    }
+  ]);
+
+  checkServicesAreMonitored([
+    {
+      name: data.services.service1.name
+    }
+  ]);
+
+  notificationEnabled = true;
+  notificationWithServices = true;
+});
+
+afterEach(() => {
+  cy.stopWebContainer();
+});
+
+Given('a user with access to the Notification Rules page', () => {
+  cy.navigateTo({
+    page: 'Notifications',
+    rootItemNumber: 3,
+    subMenu: 'Notifications'
+  });
+});
+
+Given('a Notification Rule is already created', () => {
+  createNotification(notificationBody);
+});
+
+Given('the user is on the Notification Rules page', () => {
+  cy.url().should('include', '/configuration/notifications');
+});
+
+When('the user selects the edit action on a Notification Rule', () => {
+  cy.contains(notificationBody.name).click();
+});
+
+When(
+  'the user changes the resources selection and corresponding status change parameters',
+  () => {
+    cy.contains('Include services for these hosts').click();
+    notificationWithServices = false;
+  }
+);
+
+When('the user saves to confirm the changes', () => {
+  cy.get('[data-testid="Save"]').click();
+});
+
+When('the notification refresh delay has been reached', () => {
+  cy.submitResults([
+    {
+      host: data.hosts.host1.name,
+      output: 'submit_status_1',
+      status: 'down'
+    },
+    {
+      host: data.hosts.host1.name,
+      output: 'submit_status_2',
+      service: data.services.service1.name,
+      status: 'critical'
+    }
+  ]);
+
+  checkHostsAreMonitored([
+    {
+      name: data.hosts.host1.name,
+      status: 'down',
+      statusType: 'hard'
+    }
+  ]);
+
+  checkServicesAreMonitored([
+    {
+      name: data.services.service1.name,
+      status: 'critical'
+    }
+  ]);
+
+  waitUntilLogFileChange();
+});
+
+Then(
+  'only notifications for status changes of the updated resource parameters are sent',
+  () => {
+    if (!notificationEnabled) {
+      notificationSentCheck({
+        log: 'Notification on host 15',
+        contain: false
+      });
+      notificationSentCheck({
+        log: 'Notification on service (15,27)',
+        contain: false
+      });
+      return;
+    }
+
+    notificationSentCheck({ log: 'Notification on host 15' });
+
+    if (!notificationWithServices) {
+      notificationSentCheck({
+        log: 'Notification on service (15,27)',
+        contain: false
+      });
+      return;
+    }
+
+    notificationSentCheck({ log: 'Notification on service (15,27)' });
+  }
+);
+
+When('the user changes the contact configuration', () => {
+  cy.get('#Searchcontacts').click();
+  cy.wait('@getUsers');
+  cy.get('li > div > span.MuiButtonBase-root').eq(0).click();
```

Comment:
I think you need to find another selector because this one could be modified at any time, try looking for a test-id or id or classname.